### PR TITLE
Add pc.SceneRegistry to enable looking up scene files by their name

### PR DESCRIPTION
**Pull request by daredevildave**
Add pc.Application#getSceneUrl so that there is no need to hardcode scene urls anymore.

### CODE REVIEW

File: src/framework/application.js [hunk 1]
```diff
@@ -630,6 +634,22 @@ pc.extend(pc, function () {
...
         },
 
+        /**
+         * @function
+         * @name pc.Application#getSceneUrl
+         * @description Look up the URL of the scene hierarchy file via the name given to the scene in the editor. Use this to in {@link pc.Application#loadSceneHierarchy}.
+         * @param  {String} name The name of the scene file given in the Editor
+         * @return {String}      The URL of the scene file
+         */
+        getSceneUrl: function (name) {
```
Comment (vkalpias) [inline src/framework/application.js]: Should this live in `app.scenes` instead?

File: src/framework/scene-registry.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
+pc.extend(pc, function () {
+
+    /**
+     * @private
+     * @name pc.SceneRegistryItem
+     * @class Item to be stored in the {@link pc.SceneRegistry}
+     * @param {String} name The name of the scene
```
Comment (vkalpias) [inline src/framework/scene-registry.js]: Users can rename scenes in the Editor. Do we need to handle it?

File: src/framework/application.js (hunk 1 above)
Comment (daredevildave) [reply src/framework/application.js]: I think that would be confusing given that we already have `app.scene` as part of the renderer.

File: src/framework/scene-registry.js (hunk 2 above)
Comment (daredevildave) [reply src/framework/scene-registry.js]: Changing scene name in the editor requires a page refresh to be picked up

**Pull request merged**